File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = []
        used = [False] * len(strs)  # Track which words we've already grouped
        
        for i in range(len(strs)):
            # Skip if we already grouped this word
            if used[i]:
                continue
            
            # Start a new group with this word
            current_group = [strs[i]]
            used[i] = True
            
            # Compare this word with every other word
            for j in range(i + 1, len(strs)):
                # Skip if already grouped
                if used[j]:
                    continue
                
                # Check if strs[i] and strs[j] are anagrams
                if self.areAnagrams(strs[i], strs[j]):
                    current_group.append(strs[j])
                    used[j] = True
            
            result.append(current_group)
        
        return result
    
    def areAnagrams(self, word1: str, word2: str) -> bool:
        # Different lengths = not anagrams
        if len(word1) != len(word2):
            return False
        
        # Count characters in word1
        for char in word1:
            # Count how many times this char appears in both words
            count1 = 0
            count2 = 0
            for c in word1:
                if c == char:
                    count1 += 1
            for c in word2:
                if c == char:
                    count2 += 1
            
            # If counts don't match, not anagrams
            if count1 != count2:
                return False
        
        return True

File: Data Structures & Algorithms/n-th-tribonacci-number/submission-1.java
class Solution {
    public int tribonacci(int n) {
        if(n==0)
        {
            return 0;
        }
        else if(n==1)
        {
            return 1;
        }

        int n1=0;
        int n2=1;
        int n3=1;
        for(int i = 0;i<n-1;i++)
        {
            int n4=n1+n2+n3;
            n1=n2;
            n2=n3;
            n3=n4;
        }

        return n2;
    }
}